packages/wpcom-template-parts/src/universal-footer-navigation/index.tsx

```diff
@@ -2,28 +2,42 @@
 import {
 	useLocalizeUrl,
 	removeLocaleFromPathLocaleInFront,
+	retrieveLocaleFromPathLocaleInFront,
 	useIsEnglishLocale,
 	useLocale,
+	localizeUrl as pureLocalizeUrl,
 } from '@automattic/i18n-utils';
 import { __ } from '@wordpress/i18n';
 import SocialLogo from 'social-logos';
 import useAutomatticBrandingNoun from '../hooks/use-automattic-branding-noun';
-import { FooterProps } from '../types';
+import { FooterProps, PureFooterProps } from '../types';
 
 import './style.scss';
 
-const UniversalNavbarFooter = ( {
-	isLoggedIn = false,
-	currentRoute,
-	additonalCompanyLinks,
-	onLanguageChange,
-}: FooterProps ) => {
-	const localizeUrl = useLocalizeUrl();
-	const locale = useLocale();
-	const isEnglishLocale = useIsEnglishLocale();
-	const pathNameWithoutLocale =
-		currentRoute && removeLocaleFromPathLocaleInFront( currentRoute ).slice( 1 );
-	const automatticBranding = useAutomatticBrandingNoun();
+const defaultOnLanguageChange: React.ChangeEventHandler< HTMLSelectElement > = ( event ) => {
+	const newURL = `${ event.target.value }${ removeLocaleFromPathLocaleInFront(
+		window.location.pathname
+	) }`;
+	window.location.href = newURL;
+};
+
+/**
+ * This component doesn't depend on any hooks or state. To it's Gutenberg save.js friendly.
+ */
+export const PureUniversalNavbarFooter = ( {
+	isLoggedIn = typeof window !== 'undefined'
+		? document.body.classList.contains( 'logged-in' )
+		: false,
+	additionalCompanyLinks = null,
+	onLanguageChange = defaultOnLanguageChange,
+	localizeUrl = pureLocalizeUrl,
+	automatticBranding = {
+		article: __( 'An', __i18n_text_domain__ ),
+		noun: __( 'thingamajig', __i18n_text_domain__ ),
+	},
+}: PureFooterProps ) => {
+	const locale = retrieveLocaleFromPathLocaleInFront( window.location.pathname );
```

Comment:
It seems it doesn't work as expected in some cases - e.g., while editing the test post on `wordpress.com/es/support`, I noticed that the locale slug is not in the pathname, because the editor is loaded from `es.support.wordpress.com`.

Could we instead use the locale slug from the loaded translation data, e.g. `getLocaleData( 'happy-blocks' )?.['']?.language`?